tests/Unit/Informer/Test_Verbosity.cpp

```diff
@@ -0,0 +1,46 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "tests/Unit/TestingFramework.hpp"
+
+#include <string>
+
+#include "Informer/Tags.hpp"
+#include "Informer/Verbosity.hpp"
+#include "Options/ParseOptions.hpp"
+#include "Utilities/GetOutput.hpp"
+#include "Utilities/TMPL.hpp"
+
+namespace {
+
+void test_construct_from_options() noexcept {
+  Options<tmpl::list<OptionTags::Verbosity>> opts("");
+  opts.parse("Verbosity: Verbose\n");
+  CHECK(opts.get<OptionTags::Verbosity>() == Verbosity::Verbose);
+}
+
+void test_construct_from_options_fail() {
+  Options<tmpl::list<OptionTags::Verbosity>> opts("");
+  opts.parse("Verbosity: Braggadocious\n");  // Meant to fail.
+  opts.get<OptionTags::Verbosity>();
+}
+
+void test_ostream() noexcept {
+  CHECK(get_output(Verbosity::Silent) == "Silent");
+  CHECK(get_output(Verbosity::Quiet) == "Quiet");
+  CHECK(get_output(Verbosity::Verbose) == "Verbose");
+  CHECK(get_output(Verbosity::Debug) == "Debug");
+}
+
+}  // namespace
+
+SPECTRE_TEST_CASE("Unit.Informer.Verbosity", "[Informer][Unit]") {
+  test_construct_from_options();
+  test_ostream();
+}
+
+// [[OutputRegex, Failed to convert "Braggadocious" to Verbosity]]
+SPECTRE_TEST_CASE("Unit.Informer.Verbosity.Fail", "[Informer][Unit]") {
```

Comment:
I think rather than an error test it would be possible to combine this in the other test, catch the exception and check the exception. This'll reduce our test runtime. Something like:

```
try {
  test_construct_from_options_fail();
  CHECK(false);
} catch (THE EXPECTED_ERROR_TYPE e {
  CHECK(e.msg() == "...");
} catch (...) {
  CHECK(false);
}
```

I think @fmahebert did this in one of his PRs at one point. He might remember what code it was. I think it's the root finding code.

Reply:
Look at `test_throw_exception` in `tests/Unit/TestHelpers.hpp`

Reply:
Should `create_from_yaml<Verbosity>` call PARSE_ERROR on error (as it does now) or should it raise an exception?  And if so, which one? PARSE_ERROR eventually calls std::terminate() so I don't think you can catch it.

Reply:
It throws an exception.

Reply:
Ah, I'm not sure that exception propagates far enough out, you'd have to check.

Reply:
Are you talking about the `Options_detail::propagate_context` exception?  That is caught by other stuff in the Option parser, and then ERROR is called when it gets to the top level.

Reply:
Conclusion of a call: The exception doesn't propagate far enough to be able to catch. We will leave as is.